composition-js/src/merging/merge.ts

```diff
@@ -1248,6 +1270,12 @@ class Merger {
               overriddenSubgraphASTNode,
             ));
           }
+
+          // capture an override label if it exists
```

Comment:
We might want to do some kind of validation on the `label` here (it's easier to relax validations in the future than to add them). For example, we could require that either (1) the label has to start with a letter and consist of alphanumeric+dashes+underscore(+periods+slashes+colons if we feel like it), or (2) the label has to be of the form `percent(x)` where `x` is a number between 0 and 100 inclusive (maybe something like [js-big-decimal](https://www.npmjs.com/package/js-big-decimal) could help here). If you have any other syntax in mind for `label` we could use that too.

Reply:
Good call out. I don't _think_ we need a lib for this but let me know if you think I've missed anything in how we handle floats.
[cf27bcb](https://github.com/apollographql/federation/pull/2879/commits/cf27bcbea0eea036039563dc47721d66582742ab)

Reply:
It was mainly to handle cases where JS `Number`s lose precision and we don't faithfully copy the argument from subgraphs to supergraph, but if you're using 10 digits total (1-2 before the decimal point and 8 after), then JS `Number`s should be able to faithfully represent that.